File: app/src/main/java/org/singularux/contacts/core/ContactsPermissionManagerAndroid.java
package org.singularux.contacts.core;

import android.Manifest;
import android.content.Context;
import android.content.pm.PackageManager;

import androidx.annotation.NonNull;
import androidx.core.content.ContextCompat;

import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public class ContactsPermissionManagerAndroid implements ContactsPermissionManager {

    private final Context context;

    @Override
    public boolean hasPermission(@NonNull ContactsPermission permission) {
        return ContextCompat.checkSelfPermission(context, getPermissionString(permission)) ==
                PackageManager.PERMISSION_GRANTED;
    }

    @Override
    public @NonNull String getPermissionString(@NonNull ContactsPermission permission) {
        switch (permission) {
            case READ_CONTACTS:
                return Manifest.permission.READ_CONTACTS;
            case READ_PROFILE:
                return "android.permission.READ_PROFILE";
            case WRITE_CONTACTS:
                return Manifest.permission.WRITE_CONTACTS;
            case WRITE_PROFILE:
                return "android.permission.WRITE_PROFILE";
            default:
                throw new IllegalArgumentException("Wrong permission enum: " + permission);
        }
    }

}

File: core/permission/src/main/kotlin/org/singularux/contacts/core/permission/ContactsPermissionManagerAndroid.kt
package org.singularux.contacts.core.permission

import android.Manifest
import android.content.Context
import android.content.pm.PackageManager
import androidx.core.content.ContextCompat

internal class ContactsPermissionManagerAndroid(
    private val context: Context
): ContactsPermissionManager {

    override fun hasPermission(permission: ContactsPermission): Boolean {
        return ContextCompat.checkSelfPermission(context,
            getPermissionString(permission)) == PackageManager.PERMISSION_GRANTED
    }

    override fun getPermissionString(permission: ContactsPermission): String {
        return when (permission) {
            ContactsPermission.READ_CONTACTS -> Manifest.permission.READ_CONTACTS
            ContactsPermission.WRITE_CONTACTS -> Manifest.permission.WRITE_CONTACTS
        }
    }

}